Sources/SwiftSyntax/SyntaxArena.swift

```diff
@@ -24,7 +24,7 @@ fileprivate struct AtomicBool {
   }
 }
 #else
-import _AtomicBool
+import _SwiftSyntaxCShims
 #endif
```

Comment:
If we’ve got `AtomicBool` available in the CMake build as well now (I never thought of making it header-only, nice idea), I think we can also remove the hack of defining a fake `AtomicBool` when `SWIFT_SYNTAX_BUILD_USING_CMAKE` is sent and instead always use `_SwiftSyntaxCShims`.